datalad_container/extractors/tests/test_metalad_container.py

```diff
@@ -0,0 +1,75 @@
+import os.path as op
+import pytest
+import subprocess
+import sys
+from pathlib import Path
+from shutil import which
+from unittest.mock import patch
+
+from datalad.api import (
+    clone,
+    Dataset,
+    meta_extract,
+)
+from datalad.cmd import (
+    StdOutCapture,
+    WitlessRunner,
+)
+from datalad.support.exceptions import CommandError
+from datalad.support.external_versions import external_versions, UnknownVersion
+from datalad.tests.utils_pytest import (
+    SkipTest,
+    assert_in,
+    assert_raises,
+    eq_,
+    ok_exists,
+    with_tempfile,
+    with_tree,
+)
+
+if not external_versions["datalad_metalad"]:
+    raise SkipTest("skipping metalad tests")
+
+# Must come after skiptest or imports will not work
+from datalad_container.extractors.metalad_container import MetaladContainerInspect
+
+
+@with_tempfile
```

Comment:
next two tests rely on `apptainer` to be present . ATM AFAIK our CI installs singularity... could we make it more flexible -- e.g. add some module level const "CONTAINER_SYSTEM", skip entire module if None available, and store  string either `"singularity"` or `"apptainer"` and then use it in this tests, thus avoiding hardcoding any and allowing for either to be used here.

that would also avoid possible shadowing of CalledProcessError due to simply not finding that executable instead of it unable to work with  bad path